base/loading.jl

```diff
@@ -498,14 +490,29 @@ function explicit_project_deps_get(project_file::String, name::String)::Union{Bo
                 end
             elseif state == :deps
                 if (m = match(re_key_to_string, line)) != nothing
-                    m.captures[1] == name && return UUID(m.captures[2])
+                    deps[m.captures[1]] = UUID(m.captures[2])
                 end
             elseif contains(line, re_section)
                 state = :deps
             end
         end
-        return false
     end
+    return root_name, root_uuid, deps
+end
+# find project file root or deps `name => uuid` mapping
+#  - `false` means: did not find `name`
+#  - `true` means: found `name` without UUID
```

Comment:
Maybe I'm just missing something, but I don't see how this ever returned `true` before. @StefanKarpinski